modules/service/src/main/scala/lucuma/odb/service/Services.scala

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.service
+
+import cats.effect.Concurrent
+import cats.effect.MonadCancelThrow
+import cats.effect.Resource
+import cats.effect.std.UUIDGen
+import fs2.compat.NotGiven
+import lucuma.core.model.User
+import lucuma.itc.client.ItcClient
+import lucuma.odb.logic.Generator
+import lucuma.odb.logic.Itc
+import lucuma.odb.sequence.util.CommitHash
+import natchez.Trace
+import skunk.Session
+import skunk.Transaction
+
+/** Witnesses that there is no transaction in context. */
+type NoTransaction[F[_]] = NotGiven[Transaction[F]]
```

Comment:
Hmmm, why is there an `fs2.compat.NotGiven` instead of just `scala.util.NotGiven`?

Reply:
Ah auto-import picked the wrong thing. Will fix.